src/main/java/DeadlineCommand.java

```diff
@@ -0,0 +1,53 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Class to handle commands to add a deadline to the bot's list.
+ * @author vanGoghhh
+ */
+
+public class DeadlineCommand extends Command {
+
+    private String command;
+
+    /**
+     * Constructor for a deadlinecommand object.
+     * @param command Complete line of the deadline command entered by user.
+     */
+    public DeadlineCommand(String command) {
+
+        this.command = command;
+    }
+
+    /**
+     * Method to execute entirely when a deadline command is entered by the user.
+     * @param tasks Tasklist containing all the tasks.
+     * @param dukeUI UI to print string responses by the bot.
+     * @throws InvalidTaskDescriptionException when an inaccurate task description is entered.
+     * @throws WrongDateFormatException when the date is not typed in the correct format.
```

Comment:
I like how you have separate exceptions for the different kind of errors that could have been thrown by the compiler!